sklearn/utils/__init__.py

```diff
@@ -351,7 +351,7 @@ def safe_sqr(X, copy=True):
     -------
     X ** 2 : element wise square
     """
-    X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
+    X = check_array(X, accept_sparse=['csr', 'csc', 'coo'], ensure_2d=False)
```

Comment:
This is mainly checking for sparse formats and finiteness. Since the `_ensure_sparse_format` is an internal function, I have kept the `check_array` call as it is.